src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -462,6 +465,75 @@ s.searchParameterInfo.Name switch
             }
         }
 
+        private static void RemoveRedundantParameters(IndentedStringBuilder stringBuilder, SqlParameterCollection sqlParameterCollection)
+        {
+            var commandText = stringBuilder.ToString();
+            if (!commandText.Contains("cte", StringComparison.OrdinalIgnoreCase))
+            {
+                commandText = commandText.Replace("DISTINCT ", string.Empty, StringComparison.OrdinalIgnoreCase);
+                if (!commandText.Contains(" OR ", StringComparison.OrdinalIgnoreCase))
+                {
+                    commandText = RemoveRedundantComparisons(commandText, sqlParameterCollection, '>');
+                    commandText = RemoveRedundantComparisons(commandText, sqlParameterCollection, '<');
+                }
+            }
+
+            stringBuilder.Clear();
+            stringBuilder.Append(commandText);
+        }
+
+        private static string RemoveRedundantComparisons(string commandText, SqlParameterCollection sqlParameterCollection, char operatorChar)
```

Comment:
Everything looks good so far. 
My only concern: should we add a new unit test passing a query which contains the expression that must be removed, and then compare with the expected output? 

Maybe moving the logic of methods RemoveRedundantParameters and  RemoveRedundantComparisons to a static class can help us making those tests easier to be written.

Reply:
The methods now have their own class, and there are unit tests for the simplification steps